homeassistant/helpers/entity_platform.py

```diff
@@ -52,22 +51,25 @@ def __init__(self, *, hass, logger, domain, platform_name, platform,
         # which powers entity_component.add_entities
         if platform is None:
             self.parallel_updates = None
+            self.parallel_updates_semaphore = None
             return
 
-        # Async platforms do all updates in parallel by default
-        if hasattr(platform, 'async_setup_platform'):
-            default_parallel_updates = 0
-        else:
-            default_parallel_updates = 1
-
-        parallel_updates = getattr(platform, 'PARALLEL_UPDATES',
-                                   default_parallel_updates)
-
-        if parallel_updates:
-            self.parallel_updates = asyncio.Semaphore(
-                parallel_updates, loop=hass.loop)
-        else:
-            self.parallel_updates = None
+        self.parallel_updates = getattr(platform, 'PARALLEL_UPDATES', None)
+        # semaphore will be created on demand
+        self.parallel_updates_semaphore = None
+
+    def _get_parallel_updates_semaphore(self):
+        """Get or create a semaphore for parallel updates."""
+        if self.parallel_updates_semaphore is None:
+            # Even if self.parallel_updates is 0 or None, we still create
+            # a semaphore its default value is 1. The decision whether
+            # apply this semaphore to entity is in _async_add_entity() base on
```

Comment:
If a component defines a parallel updates value, we should always honor it.

Reply:
So shouldn't we just return None in that case ? Especially because this method would never be called?

Reply:
Outdated comment, this function would not be called if unnecessary.

Reply:
Probably I should remove this function as whole

Reply:
No, we need keep this function, it is important to share the semaphore cross entities. The code shall be good as current.